src/encode.cpp

```diff
@@ -381,6 +381,70 @@ optional<string> encodeOp(State &st, mlir::tensor::FromElementsOp op) {
   return {};
 }
 
+template<>
+optional<string> encodeOp(State &st, mlir::tensor::ExtractSliceOp op) {
+  vector<Expr> offsets, sizes, strides;
+  auto src = st.regs.get<Tensor>(op.getOperand(0));
+  auto srcType = op.getOperand(0).getType().dyn_cast<mlir::ShapedType>();
+  auto res = op.getResult();
+  auto resType = res.getType().dyn_cast<mlir::ShapedType>();
+
+  auto varIdx = 0;
+  for (auto s: extractFromI64ArrayAttr(op.static_offsets())){
+    if(s ==-0x8000000000000000)
```

Comment:
a space after `if`
BTW, where did you find this check (a comparison with `-0x8000000000000000`)?

Reply:
found a simple function for reading inputs